store/src/sumtree.rs

```diff
@@ -306,6 +306,13 @@ where
 			return None;
 		}
 
+		if let Some(rewind) = self.rewind {
+			println!("sumtree backed get: rewind exists {:?}, position {}", rewind, position);
+			if rewind.0 < position {
```

Comment:
I don't know if this is the right thing to do but here we short-circuit and return None if we are are trying to get beyond the rewound position.